rules/sort-intersection-types.ts

```diff
@@ -117,69 +118,48 @@ export default createEslintRule<Options, MESSAGE_ID>({
         sourceCode,
       })
 
+      let typeToGroupMap: Record<string, Group> = {
```

Comment:
[suggestion]

For stronger typing:

```ts
let typeToGroupMap: {[key in TSESTree.TypeNode['type']]?: Group} = {
```

Performance wise, I think that `switch-case` statements and object literals are similar (`O(1)`). 

Readability-wise, I prefer the `switch-case` version because it shows which cases are grouped. The object literal version is not sorted by values here, so while it's theoretically easier to read when searching by type, I think it's handier to regroup by `group` instead. It's just my opinion though.

Reply:
Back to the previous version.